dbt/models/reporting/reporting.ratio_stats.py

```diff
@@ -1,157 +1,129 @@
 )
 
-import numpy as np  # noqa: I001 E402
-import pandas as pd  # noqa: E402
-from assesspy import (  # noqa: E402
-    boot_ci,  # noqa: E402
-    cod,  # noqa: E402
-    cod_ci as cod_boot,  # noqa: E402
-    cod_met,  # noqa: E402
-    mki,  # noqa: E402
-    mki_met,  # noqa: E402
-    prb,  # noqa: E402
-    prb_met,  # noqa: E402
-    prd,  # noqa: E402
-    prd_met,  # noqa: E402
-    prd_ci as prd_boot,  # noqa: E402
-)
-
-
-def median_boot(ratio, nboot=100, alpha=0.05):
-    return boot_ci(np.median, nboot=nboot, alpha=alpha, ratio=ratio)
-
-
-def ccao_median(x):
-    # Remove top and bottom 5% of ratios as per CCAO Data Department SOPs
-    no_outliers = x.between(
-        x.quantile(0.05), x.quantile(0.95), inclusive="neither"
-    )
-
-    x_no_outliers = x[no_outliers]
-
-    median_n = sum(no_outliers)
-
-    median_val = np.median(x_no_outliers)
-    median_ci = median_boot(x_no_outliers, nboot=1000)
-    median_ci = f"{median_ci[0]}, {median_ci[1]}"
-
-    out = [median_val, median_ci, median_n]
-
-    return out
-
-
-def ccao_mki(fmv, sale_price):
-    ratio = fmv / sale_price
-
-    # Remove top and bottom 5% of ratios as per CCAO Data Department SOPs
-    no_outliers = ratio.between(
-        ratio.quantile(0.05), ratio.quantile(0.95), inclusive="neither"
-    )
-
-    fmv_no_outliers = fmv[no_outliers]
-    sale_price_no_outliers = sale_price[no_outliers]
-
-    mki_n = sum(no_outliers)
-
-    if mki_n >= 20:
-        mki_val = mki(fmv_no_outliers, sale_price_no_outliers)
-        met = mki_met(mki_val)
-
-        out = [mki_val, met, mki_n]
-
-    else:
-        out = [None, None, mki_n]
-
-    return out
 
+def ccao_drop_outliers(
+    estimate: Union[list[int], list[float], pd.Series],
+    sale_price: Union[list[int], list[float], pd.Series],
+) -> tuple[pd.Series, pd.Series, float]:
+    """
+    Helper function to drop the top and bottom N% (usually 5%) of the input
+    ratios, per CCAO SOPs and IAAO recommendation.
+    """
+    ratio: pd.Series = estimate / sale_price
+    ratio_not_outlier = ratio.between(
+        ratio.quantile(CCAO_LOWER_QUANTILE),
+        ratio.quantile(CCAO_UPPER_QUANTILE),
+        inclusive="neither",
+    ).reset_index(drop=True)
 
-def ccao_cod(ratio):
-    # Remove top and bottom 5% of ratios as per CCAO Data Department SOPs
-    no_outliers = ratio[
-        ratio.between(
-            ratio.quantile(0.05), ratio.quantile(0.95), inclusive="neither"
-        )
-    ]
+    estimate_no_outliers = estimate[ratio_not_outlier]
+    sale_price_no_outliers = sale_price[ratio_not_outlier]
+    n: float = float(estimate_no_outliers.size)
 
-    cod_n = no_outliers.size
+    return estimate_no_outliers, sale_price_no_outliers, n
 
-    if cod_n >= 20:
-        cod_val = cod(no_outliers)
-        cod_ci = cod_boot(no_outliers, nboot=1000)
-        cod_ci = f"{cod_ci[0]}, {cod_ci[1]}"
-        met = cod_met(cod_val)
-        out = [cod_val, cod_ci, met, cod_n]
 
+def ccao_metric(
```

Comment:
I condensed the individual metric functions into this wrapper. It:

- Drops any outliers, per `ccao_drop_outliers`
- Checks the min sample size requirement
- Calculates the stat (COD, PRD, PRB, or MKI)
- Calculates the CI for the stat (except for MKI)
- Calculates the post-outlier-drop sample size (n)
- Returns a dictionary for use in `calc_summary`

We _could_ move this and the other `ccao_` prefixed functions into the `ccao` package, but I'm honestly not sure it's necessary. This function and the others are sufficiently short that it doesn't seem like a big deal to have them in here.